src/OSPSuite.Presentation/Presenters/SimulationOutputMappingPresenter.cs

```diff
@@ -127,24 +140,64 @@ private IEnumerable<DataRepository> allAvailableObservedDataUsedBySimulation()
       public void UpdateSimulationOutputMappings(SimulationOutputMappingDTO simulationOutputMappingDTO)
       {
          MarkSimulationAsChanged();
+
+         if (simulationOutputMappingDTO.Output.DisplayString.Equals(Captions.SimulationUI.NoneEditorNullText))
+         {
+            removeOutputMapping(simulationOutputMappingDTO);
+            return;
+         }
+
          if (!_simulation.OutputMappings.OutputMappingsUsingDataRepository(simulationOutputMappingDTO.ObservedData).Any())
             _simulation.OutputMappings.Add(simulationOutputMappingDTO.Mapping);
 
          simulationOutputMappingDTO.Scaling = _outputMappingMatchingTask.DefaultScalingFor(simulationOutputMappingDTO.Output.Quantity);
       }
 
+      private void removeOutputMapping(SimulationOutputMappingDTO simulationOutputMappingDTO)
+      {
+         simulationOutputMappingDTO.Output = null;
+         _simulation.OutputMappings.Remove(simulationOutputMappingDTO.Mapping);
+         _view.RefreshGrid();
+      }
+
       public void MarkSimulationAsChanged()
       {
          _simulation.HasChanged = true;
       }
 
-      public void RemoveOutputMapping(SimulationOutputMappingDTO outputMappingDTO)
+      public void RemoveObservedData(SimulationOutputMappingDTO outputMappingDTO)
       {
-         outputMappingDTO.Output = null;
+         var parameterIdentifications = findParameterIdentificationsUsing(outputMappingDTO.ObservedData).ToList();
```

Comment:
Why is this done in a presneter? This looks like code that should be done at the task level. This presenter has nothing to do with ParameterIdentification. Why is this dealing with this concerns suddenly?

Reply:
You are right, I have overlooked this and I totally agree. Then the task will also be used in PK-Sim and MoBi - since we are going to keep the existing way of removing observed data from simulations.

Reply:
this was moved to the service